### ISSUE TITLE
L034 should report the warning on the earlier, more complex select target

### ISSUE THREAD

**Issue by pwildenhain** [sqlfluff/sqlfluff#916]
@pwildenhain: Thanks for looking at this. I am seeing the same behavior. I think the reason for this is that the "noqa" filter applies based on where the line rule **reports** the error. Notice that the two L034 errors are reported at "L: 7" and L: 10", the lines with `field3` and `field4`.

if you look at the implementation of L034, you'll see why -- it reports the warning when it finds a **simple** element following a more complex one, and the anchor (which determines the line and position of the warning) is the later, simpler segment:
https://github.com/sqlfluff/sqlfluff/blob/master/src/sqlfluff/core/rules/std/L034.py#L38LL42

As a test, I moved the "disable/enable L034" code around line 7 (rather than line 5), and it seems to behave as expected (the L034 report on line 7 goes away).

```
SELECT
    FIELD_1,
    FIELD_2,
    max(field_1) as field_1_max,
    -- noqa: disable=L034
    field_3,
    -- noqa: enable=L034
    -- noqa: disable=L009,L010,L014
    max(field_3) as max_field_3,
    field_4
from MY_TABLE
WHERE
    field_1 > 1 and field_2 > 2 and field_3 > 3
```

Output:
```
L:   4 | P:   9 | L014 | Inconsistent capitalisation of unquoted identifiers.
L:   4 | P:  18 | L010 | Inconsistent capitalisation of keywords.
L:   4 | P:  21 | L014 | Inconsistent capitalisation of unquoted identifiers.
L:   6 | P:   5 | L014 | Inconsistent capitalisation of unquoted identifiers.
L:  10 | P:   5 | L034 | Use wildcards then simple select targets before
                       | calculations and aggregates.
```

I think your point is a good one -- now that I'm looking at this, perhaps L034 should report the warning on the earlier, more complex select target. I think that should be a separate issue and PR from this work.

_Originally posted by @barrywhart in https://github.com/sqlfluff/sqlfluff/issues/891#issuecomment-810636517_
